service-bus.api.md

```diff
@@ -0,0 +1,337 @@
+  receiveBatch(maxMessages: number, options?: ReceiveBatchOptions): Promise<ReceivedMessageT[]>;
+  receiveDeferredMessage(
+    sequenceNumber: Long,
+    options?: OperationOptions
+  ): Promise<ReceivedMessageT | undefined>;
+  receiveDeferredMessages(
+    sequenceNumbers: Long[],
+    options?: OperationOptions
+  ): Promise<ReceivedMessageT[]>;
+  receiveMode: "peekLock" | "receiveAndDelete";
+  subscribe(handlers: MessageHandlers<ReceivedMessageT>, options?: SubscribeOptions): void;
+}
+
+export { RetryOptions };
+
+// @public
+export interface RuleDescription {
+  action?: string;
+  filter?: string | CorrelationFilter;
+  name: string;
+}
+
+// @public
+export interface Sender {
+  cancelScheduledMessage(sequenceNumber: Long): Promise<void>;
+  cancelScheduledMessages(sequenceNumbers: Long[]): Promise<void>;
+  close(): Promise<void>;
+  createBatch(options?: CreateBatchOptions): Promise<ServiceBusMessageBatch>;
+  isClosed: boolean;
+  scheduleMessage(scheduledEnqueueTimeUtc: Date, message: ServiceBusMessage): Promise<Long>;
+  scheduleMessages(scheduledEnqueueTimeUtc: Date, messages: ServiceBusMessage[]): Promise<Long[]>;
+  send(message: ServiceBusMessage): Promise<void>;
+  sendBatch(messageBatch: ServiceBusMessageBatch): Promise<void>;
+}
+
+// @public
+export class ServiceBusClient {
+  constructor(connectionString: string, options?: ServiceBusClientOptions);
+  constructor(
+    hostName: string,
+    tokenCredential: TokenCredential,
+    options?: ServiceBusClientOptions
+  );
+  close(): Promise<void>;
+  getDeadLetterReceiver(
+    queueName: string,
+    receiveMode: "peekLock",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessageWithLock>;
+  getDeadLetterReceiver(
+    queueName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessage>;
+  getDeadLetterReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "peekLock",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessageWithLock>;
+  getDeadLetterReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessage>;
+  getReceiver(
+    queueName: string,
+    receiveMode: "peekLock",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessageWithLock>;
+  getReceiver(
+    queueName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessage>;
+  getReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "peekLock",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessageWithLock>;
+  getReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessage>;
+  getSender(queueOrTopicName: string, options?: GetSenderOptions): Sender;
+  getSessionReceiver(
```

Comment:
We have feedback for this to be async. We need issue to talk about getReceiver and getSender to be async as well